### ISSUE TITLE
Ask about the af2 dependency under RFDesign, 'RFDesign/scripts/af2_metrics.py' 

### ISSUE THREAD

**Issue by Licko0909** [RosettaCommons/RFDesign#7]
Hello dear author! 

I'm very interested in your work, it's a great job! 

So, I've recently been designing new sequences with RFDesign. In the process, I tried to execute `RFDesign/scripts/af2_metrics.py` to score the results using Alphafold2, but I ran into some problems: the process requires the AF2 dependency to be installed in the `RFDesign` environment, and I tried to install the AF2 dependency on top of RFDesign base to install the alphafold2 environment, but it always fails to execute `af2_metrics.py` successfully, the result is as follows:

![image](https://user-images.githubusercontent.com/27897166/173558227-b16544d3-275c-463b-bcaa-e5dc7bac20c1.png)

![image](https://user-images.githubusercontent.com/27897166/173558000-ca6e5aaa-255b-40da-9527-67a0d10ce55d.png)

I personally think it is the problem of `jax` and` tensorflow` and `cuda` version mismatch. What I need to ask is, what version of `jax` and `python` are you using and can you expose the files with RFDesign and alphafold2 dependencies in them? For example `xx.yml` or `requirement.txt`

This is my `conda list`:

>
> Name                    Version                   Build  Channel
>_libgcc_mutex             0.1                        main    conda-forge
>_openmp_mutex             4.5                       1_gnu    defaults
_tflow_select             2.1.0                       gpu    defaults
abseil-cpp                20210324.0           h9c3ff4c_0    conda-forge
absl-py                   1.1.0              pyhd8ed1ab_0    conda-forge
aiohttp                   3.7.4.post0      py38h497a2fe_0    conda-forge
astor                     0.8.1              pyh9f0ad1d_0    conda-forge
astunparse                1.6.3              pyhd8ed1ab_0    conda-forge
async-timeout             3.0.1                   py_1000    conda-forge
attrs                     21.4.0             pyhd8ed1ab_0    conda-forge
biopython                 1.79                     pypi_0    pypi
blas                      1.0                         mkl    conda-forge
blinker                   1.4                        py_1    conda-forge
brotlipy                  0.7.0           py38h27cfd23_1003    defaults
bzip2                     1.0.8                h7b6447c_0    defaults
c-ares                    1.17.1               h7f98852_1    conda-forge
ca-certificates           2022.5.18.1          ha878542_0    conda-forge
cachetools                4.2.4              pyhd8ed1ab_0    conda-forge
certifi                   2022.5.18.1      py38h578d9bd_0    conda-forge
cffi                      1.14.6           py38h400218f_0    defaults
chardet                   4.0.0            py38h578d9bd_3    conda-forge
charset-normalizer        2.0.4              pyhd3eb1b0_0    defaults
chex                      0.0.7                    pypi_0    pypi
click                     8.1.3            py38h578d9bd_0    conda-forge
colorama                  0.4.4              pyh9f0ad1d_0    conda-forge
contextlib2               21.6.0                   pypi_0    pypi
cryptography              36.0.0           py38h9ce1e76_0    defaults
cudatoolkit               11.3.1               h2bc3f7f_2    defaults
dgl-cuda11.3              0.7.2                    py38_0    dglteam
dm-haiku                  0.0.4                    pypi_0    pypi
dm-tree                   0.1.6                    pypi_0    pypi
ffmpeg                    4.3                  hf484d3e_0    pytorch
fftw                      3.3.9                h27cfd23_1    defaults
freetype                  2.11.0               h70c0345_0    defaults
gast                      0.3.3                      py_0    conda-forge
giflib                    5.2.1                h7b6447c_0    defaults
gmp                       6.2.1                h2531618_2    defaults
gnutls                    3.6.15               he1e5248_0    defaults
google-auth               1.35.0             pyh6c4a22f_0    conda-forge
google-auth-oauthlib      0.4.6              pyhd8ed1ab_0    conda-forge
google-pasta              0.2.0              pyh8c360ce_0    conda-forge
googledrivedownloader     0.4                pyhd3deb0d_1    conda-forge
grpc-cpp                  1.37.1               h36de60a_0    conda-forge
grpcio                    1.37.1           py38hdd6454d_0    conda-forge
h5py                      2.10.0          nompi_py38h9915d05_106    conda-forge
hdf5                      1.10.6          nompi_h7c3c948_1111    conda-forge
hhsuite                   3.3.0           py38pl5262hc37a69a_2    bioconda
hmmer                     3.3.2                h1b792b2_1    bioconda
icu                       68.1                 h58526e2_0    conda-forge
idna                      3.3                pyhd3eb1b0_0    defaults
immutabledict             2.0.0                    pypi_0    pypi
importlib-metadata        4.11.4           py38h578d9bd_0    conda-forge
intel-openmp              2021.4.0          h06a4308_3561    defaults
jax                       0.3.13                   pypi_0    pypi
jaxlib                    0.3.10                   pypi_0    pypi
jinja2                    3.0.3              pyhd8ed1ab_0    conda-forge
joblib                    1.1.0              pyhd8ed1ab_0    conda-forge
jpeg                      9d                   h7f8727e_0    defaults
kalign2                   2.04                 h779adbc_2    bioconda
keras-preprocessing       1.1.2              pyhd8ed1ab_0    conda-forge
krb5                      1.19.2               hcc1bbae_0    conda-forge
lame                      3.100                h7b6447c_0    defaults
lcms2                     2.12                 h3be6417_0    defaults
ld_impl_linux-64          2.35.1               h7274673_9    defaults
libcurl                   7.78.0               h2574ce0_0    conda-forge
libedit                   3.1.20191231         he28a2e2_2    conda-forge
libev                     4.33                 h516909a_1    conda-forge
libffi                    3.3                  he6710b0_2    defaults
libgcc-ng                 9.3.0               h5101ec6_17    defaults
libgfortran-ng            7.5.0               ha8ba4b0_17    defaults
libgfortran4              7.5.0               ha8ba4b0_17    defaults
libgomp                   9.3.0               h5101ec6_17    defaults
libiconv                  1.15                 h63c8f33_5    defaults
libidn2                   2.3.2                h7f8727e_0    defaults
libnghttp2                1.43.0               h812cca2_0    conda-forge
libpng                    1.6.37               hbc83047_0    defaults
libprotobuf               3.15.8               h780b84a_0    conda-forge
libssh2                   1.9.0                ha56f1ee_6    conda-forge
libstdcxx-ng              9.3.0               hd4cf53a_17    defaults
libtasn1                  4.16.0               h27cfd23_0    defaults
libtiff                   4.2.0                h85742a9_0    defaults
libunistring              0.9.10               h27cfd23_0    defaults
libuv                     1.40.0               h7b6447c_0    defaults
libwebp                   1.2.0                h89dd481_0    defaults
libwebp-base              1.2.0                h27cfd23_0    defaults
lie-learn                 0.0.1.post1              pypi_0    pypi
lz4-c                     1.9.3                h295c915_1    defaults
markdown                  3.3.7              pyhd8ed1ab_0    conda-forge
markupsafe                2.0.1            py38h497a2fe_0    conda-forge
mkl                       2021.4.0           h06a4308_640    defaults
mkl-service               2.4.0            py38h7f8727e_0    defaults
mkl_fft                   1.3.1            py38hd3c417c_0    defaults
mkl_random                1.2.2            py38h51133e4_0    defaults
ml-collections            0.1.0                    pypi_0    pypi
multidict                 5.1.0            py38h497a2fe_1    conda-forge
ncurses                   6.3                  h7f8727e_2    defaults
nettle                    3.7.3                hbbd107a_1    defaults
networkx                  2.6.3              pyhd3eb1b0_0    defaults
numpy                     1.21.2           py38h20f2e39_0    defaults
numpy-base                1.21.2           py38h79a1101_0    defaults
oauthlib                  3.2.0              pyhd8ed1ab_0    conda-forge
ocl-icd                   2.3.0                h7f98852_0    conda-forge
ocl-icd-system            1.0.0                         1    conda-forge
olefile                   0.46               pyhd3eb1b0_0    defaults
openh264                  2.1.1                h4ff587b_0    defaults
openmm                    7.5.1            py38ha082873_1    conda-forge
openssl                   1.1.1o               h7f8727e_0    defaults
opt_einsum                3.3.0              pyhd8ed1ab_1    conda-forge
packaging                 21.3               pyhd3eb1b0_0    defaults
pandas                    1.3.1            py38h1abd341_0    conda-forge
pdbfixer                  1.7                pyhd3deb0d_0    conda-forge
perl                      5.26.2               h14c3975_0    defaults
pillow                    8.4.0            py38h5aabda8_0    defaults
pip                       21.2.4           py38h06a4308_0    defaults
protobuf                  3.15.8           py38h709712a_0    conda-forge
pyasn1                    0.4.8                      py_0    conda-forge
pyasn1-modules            0.2.7                      py_0    conda-forge
pycparser                 2.21               pyhd3eb1b0_0    defaults
pyg                       2.0.3           py38_torch_1.10.0_cu113    pyg
pyjwt                     2.4.0              pyhd8ed1ab_0    conda-forge
pyopenssl                 21.0.0             pyhd3eb1b0_1    defaults
pyparsing                 3.0.4              pyhd3eb1b0_0    defaults
pysocks                   1.7.1            py38h06a4308_0    defaults
python                    3.8.12               h12debd9_0    defaults
python-dateutil           2.8.2              pyhd8ed1ab_0    conda-forge
python-flatbuffers        1.12               pyhd8ed1ab_1    conda-forge
python-louvain            0.15               pyhd3deb0d_0    conda-forge
python_abi                3.8                      2_cp38    conda-forge
pytorch                   1.10.1          py3.8_cuda11.3_cudnn8.2.0_0    pytorch
pytorch-cluster           1.5.9           py38_torch_1.10.0_cu113    pyg
pytorch-mutex             1.0                        cuda    pytorch
pytorch-scatter           2.0.9           py38_torch_1.10.0_cu113    pyg
pytorch-sparse            0.6.12          py38_torch_1.10.0_cu113    pyg
pytorch-spline-conv       1.2.1           py38_torch_1.10.0_cu113    pyg
pytz                      2021.3             pyhd8ed1ab_0    conda-forge
pyu2f                     0.1.5              pyhd8ed1ab_0    conda-forge
pyyaml                    5.4.1            py38h497a2fe_0    conda-forge
re2                       2021.04.01           h9c3ff4c_0    conda-forge
readline                  8.1                  h27cfd23_0    defaults
requests                  2.26.0             pyhd3eb1b0_0    defaults
requests-oauthlib         1.3.1              pyhd8ed1ab_0    conda-forge
rsa                       4.8                pyhd8ed1ab_0    conda-forge
scikit-learn              1.0.1            py38h51133e4_0    defaults
scipy                     1.7.3            py38hc147768_0    defaults
setuptools                58.0.4           py38h06a4308_0    defaults
six                       1.16.0             pyhd3eb1b0_0    defaults
snappy                    1.1.8                he1b5a44_3    conda-forge
sqlite                    3.37.0               hc218d9a_0    defaults
svgwrite                  1.4.2                    pypi_0    pypi
tabulate                  0.8.9                    pypi_0    pypi
tensorboard               2.4.1              pyhd8ed1ab_1    conda-forge
tensorboard-plugin-wit    1.8.1              pyhd8ed1ab_0    conda-forge
tensorflow                2.4.1            py38h578d9bd_0    conda-forge
tensorflow-base           2.4.1            py38h83f5f1d_0    conda-forge
tensorflow-estimator      2.4.0              pyh9656e83_0    conda-forge
tensorflow-gpu            2.4.1                h30adc30_0    defaults
termcolor                 1.1.0                      py_2    conda-forge
threadpoolctl             3.0.0              pyh8a188c0_0    conda-forge
tk                        8.6.11               h1ccaba5_0    defaults
toolz                     0.11.2                   pypi_0    pypi
torchaudio                0.10.1               py38_cu113    pytorch
torchvision               0.11.2               py38_cu113    pytorch
tqdm                      4.62.3             pyhd8ed1ab_0    conda-forge
tree                      0.2.4                    pypi_0    pypi
typing-extensions         3.10.0.2             hd3eb1b0_0    defaults
typing_extensions         3.10.0.2           pyh06a4308_0    defaults
urllib3                   1.26.7             pyhd3eb1b0_0    defaults
werkzeug                  2.1.2              pyhd8ed1ab_1    conda-forge
wheel                     0.37.0             pyhd3eb1b0_1    defaults
wrapt                     1.12.1           py38h497a2fe_3    conda-forge
xz                        5.2.5                h7b6447c_0    defaults
yacs                      0.1.6                      py_0    conda-forge
yaml                      0.2.5                h516909a_0    conda-forge
yarl                      1.6.3            py38h497a2fe_2    conda-forge
zipp                      3.8.0              pyhd8ed1ab_0    conda-forge
zlib                      1.2.11               h7f8727e_4    defaults
zstd                      1.4.9                haebb681_0    defaults

I would really like to get your answer, thank you very much!

**Comment by jueseph** [comment]
oh, here is also the tensorflow version

tensorflow                2.5.0                    pypi_0    pypi

**Comment by Licko0909** [comment]
Thank you very much! Looking forward to your updates

**Comment by jueseph** [comment]
Here are the versions of jax, python, cuda, and pytorch:

jax                       0.2.17                   pypi_0    pypi
jaxlib                    0.1.69+cuda110           pypi_0    pypi
python                    3.8.2               hcff3b4d_14
pytorch                   1.7.0           py3.8_cuda10.2.89_cudnn7.6.5_0
 pytorch
cudatoolkit               10.2.89              hfd86e86_1

we actually run this on a few different environments with slightly
different versions of things and it generally works. the SE3.yml file in
the repo is an export of one of our environments used to run
hallucination/inpainting/scoring scripts, but now I see that it doesn't
include jax for some reason.

we are about to make a major update to this repo soon. i'll try to fix this
issue along with that.
